api/commands/build.ts

```diff
@@ -82,9 +82,13 @@ export class Builder {
     const tasks = t.disabled
       ? emptyTasks
       : emptyTasks.concat(
-          (t.preOps || []).map(pre => ({ statement: pre })),
+          t.incrementalQuery
```

Comment:
- we shouldn't check whether the table is incremental or not using the existence of this field; old versions of `@dataform/core` don't know how to set it. we should instead use `t.type === "incremental"`
- i'm not sure this is quite right... what about old versions of `@dataform/core`? we need to be backwards compatible with those; in which case `incPreOps` and `incPostOps` won't be set, and we should continue to use `preOps` and `postOps`

Reply:
Ah nice catch. I've added a check for `t.incrementalPreOps` to the ternary, I believe this will work but I'm not sure. Does the `@dataform/api` ever get recompiled whilst depending an older version of `@dataform/core`? The build file just says `//core` so I assume not.

Reply:
`@dataform/api` (effectively) doesn't depend on `@dataform/core`; `core` is responsible for creating `CompiledGraph` instances, and `api` acts upon those; `CompiledGraph` is (intended to be) the only interface between the two. thus, `api` must be backwards-compatible with old versions of `core`.